code/modules/economy/requisition/requisition_contracts.dm

```diff
@@ -102,24 +94,53 @@ ABSTRACT_TYPE(/datum/rc_entry/stack)
 	rc_eval(obj/item/eval_item)
 		. = ..()
 		if(rollcount >= count) return // standard skip-if-complete
-		if(!istype(eval_item)) return //if it's not an item, it's not a stackable
+		if(!istype(eval_item)) return // if it's not an item, it's not a stackable
 		if(eval_item.type == typepath || (typepath_alt && eval_item.type == typepath_alt))
 			rollcount += eval_item.amount
 			. = TRUE //let manager know passed eval item is claimed by contract
 
-//seeds, analyzed by genetic composition
+///Reagent entry. Searches for reagents in sent objects, consuming any suitable reagent containers until the quantity is satisfied.
+ABSTRACT_TYPE(/datum/rc_entry/reagent)
+/datum/rc_entry/reagent
+	entryclass = RC_REAGENT
+	///Reagents being looked for in the evaluation; can be a list of several, or just the one.
+	var/chemname = "water"
+
+	rc_eval(atom/eval_item)
+		. = ..()
+		if(rollcount >= count) return // standard skip-if-complete
+		if(eval_item.reagents)
+			var/C
+			if(islist(src.chemname))
+				for(var/chemplural in src.chemname)
```

Comment:
Naming doesn't make sense, at this point it should be a singular id.

Reply:
Variable renamed to chem_ids.